src/plugins/intel_gna/src/gna_graph_tools.hpp

```diff
@@ -776,14 +776,25 @@ inline void CNNNetworkReconnectLayer(CNNLayerPtr old_prev_layer, CNNLayerPtr new
     }
 }
 
+inline uint32_t GetDimFromFront(const InferenceEngine::SizeVector& dims, uint32_t dim) {
+    if (dim >= dims.size()) {
```

Comment:
I think that it is better avoid such implicit handling of dimensions. We should control in code what we are doing and which dimension we actually want to get. Otherwise it can lead to the problems in future.

Reply:
I agree, it would be good to simplify.
But, it was working like that before my changes, so left as is to not introduce regression.